File: iraf/unix/boot/spp/rpp/rppfor/lndict.f
      subroutine lndict
      integer sym (100), c
      integer sctabl, length
      integer posn, locn
      common /cdefio/ bp, buf (4096)
      integer bp
      integer buf
      common /cfname/ fcname (30)
      integer fcname
      common /cfor/ fordep, forstk (200)
      integer fordep
      integer forstk
      common /cgoto/ xfer
      integer xfer
      common /clabel/ label, retlab, memflg, col, logic0
      integer label
      integer retlab
      integer memflg
      integer col
      integer logic0
      common /cline/ dbgout, dbglev, level, linect (5), infile (5), fnam
     *p, fnames ( 150)
      integer dbgout
      integer dbglev
      integer level
      integer linect
      integer infile
      integer fnamp
      integer fnames
      common /cmacro/ cp, ep, evalst (500), deftbl
      integer cp
      integer ep
      integer evalst
      integer deftbl
      common /coutln/ outp, outbuf (74)
      integer outp
      integer outbuf
      common /csbuf/ sbp, sbuf(2048), smem(240)
      integer sbp
      integer sbuf
      integer smem
      common /cswtch/ swtop, swlast, swstak(1000), swvnum, swvlev, swvst
     *k(10), swinrg
      integer swtop
      integer swlast
      integer swstak
      integer swvnum
      integer swvlev
      integer swvstk
      integer swinrg
      common /ckword/ rkwtbl
      integer rkwtbl
      common /clname/ fkwtbl, namtbl, gentbl, errtbl, xpptbl
      integer fkwtbl
      integer namtbl
      integer gentbl
      integer errtbl
      integer xpptbl
      common /erchek/ ername, body, esp, errstk(30)
      integer ername
      integer body
      integer esp
      integer errstk
      integer mem( 60000)
      common/cdsmem/mem
      posn = 0
23000 if (.not.(sctabl (namtbl, sym, locn, posn) .ne. -1))goto 23001
      if (.not.(length(sym) .gt. 6))goto 23002
       call outch (99)
      call outtab
23004 if (.not.(mem (locn) .ne. -2))goto 23006
      c = mem (locn)
      call outch (c)
23005 locn = locn + 1
      goto 23004
23006 continue
      call outch (32)
      call outch (32)
      call outstr (sym)
      call outdon
23002 continue
      goto 23000
23001 continue
      return
      end
c     logic0  logical_column

File: unix/boot/spp/rpp/rpprat/lndict.r
#-h-  lndict			  678  local   12/01/80  15:54:13
# lndict - output long-name dictionary as a debugging aid
   include  defs

subroutine lndict

character sym (MAXTOK), c
integer sctabl, length
pointer posn, locn
include COMMON_BLOCKS

   posn = 0
   while (sctabl (namtbl, sym, locn, posn) != EOF)
      if (length(sym) > MAXIDLENGTH) {
	  call outch (LETC)
	  call outtab
	  for (; mem (locn) != EOS; locn = locn + 1) {
	     c = mem (locn) # kluge for people with LOGICAL*1 characters
	     call outch (c)
	  }
	  call outch (BLANK)
	  call outch (BLANK)
	  call outstr (sym)
	  call outdon
      }
   return
end
